### PR TITLE
[builder] refactor builder to utilize Dask

### PR DESCRIPTION
**Pull request by bkmartinjr**
This PR focuses on migrating the build phase to Dask (consolidation and validation will come in a later PR). Steps converted:
1. H5AD download now downloads each 64MiB slice of a file in parallel (approximately halving download time for all H5ADs)
2. obs/var scan (build step 3) is parallelized using Dask
3. X processing is parallelized with Dask. It now writes `raw` and `normalized` at the same time, removing the cost of reading `raw` twice.

Overall impact is simpler code and a ~20-25% reduction is wall-clock time to create the Census.

Other changes:
* `build_soma.build()` now incorporates the validation step.  `build_soma.validate()` has been removed. This was done to allow for the possibility of performing the consolidation and validation phases in parallel (work which will be done in a later PR)
* Remove dead code, eg., EagerIterator, array_chunker, etc.
* A new `test-build` command is added to the top-level `main()`, allowing for simpler (developer) testing
* A number of new config params added to control Dask behavior (e.g., enable/disable web dashboard)
* Functional enhancements to the AnnDataProxy introduced in #925
* Added an asynchronous consolidator which runs in parallel with X layer writes. End result is less time spent in the final consolidation pass.
* Tune normalization layer compression level to hit a better trade-off between cost (CPU time) and size reduction.
* Revise tests to match API modifications

### CODE REVIEW

**Comment by atarashansky** [comment]
This is a really cool use of Dask!! Bookmarking this PR for future reference....

**Comment by bkmartinjr** [comment]
> This is a really cool use of Dask!! Bookmarking this PR for future reference....

Happy to take review comments if you give it a close read....

File: tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/build_soma.py [hunk 1]
```diff
@@ -51,66 +53,92 @@ def prepare_file_system(args: CensusBuildArgs) -> None:
...
 
     prepare_file_system(args)
 
-    # Step 1 - get all source datasets
-    datasets = build_step1_get_source_datasets(args)
-
-    # Step 2 - create root collection, and all child objects, but do not populate any dataframes or matrices
-    root_collection = build_step2_create_root_collection(args.soma_path.as_posix(), experiment_builders)
-    gc.collect()
+    with create_dask_client(args, n_workers=cpu_count(), threads_per_worker=2):
```
[the remark this answers is not in the document]
Comment (bkmartinjr) [reply tools/cellxgene_census_builder/src/cellxgene_census_builder/build_soma/build_soma.py:77]: I don't like to add config where it isn't actually user configurable and/or needed for configuration.  In this case, it has more to do with the code, so it is not usefully configurable IMHO.  (this is also why it was not previously configurable - this code replicates previous implementation)

**Review by ebezzi — APPROVED** [review]
LGTM. Will do a second pass later (mostly for my knowledge).

**Pull request merged**